include/fmt/format-inl.h

```diff
@@ -482,8 +483,8 @@ FMT_FUNC fp get_cached_power(int min_exponent, int& pow10_exponent) {
   const uint64_t one_over_log2_10 = 0x4d104d42;  // round(pow(2, 32) / log2(10))
   int index = static_cast<int>(
       static_cast<int64_t>(
-          (min_exponent + fp::significand_size - 1) * one_over_log2_10 +
-          ((uint64_t(1) << 32) - 1)  // ceil
+          static_cast<uint64_t>(min_exponent + fp::significand_size - 1) * one_over_log2_10 +
```

Comment:
Please replace the cast with `to_unsigned`.

Reply:
Your suggested change causes four tests failures (building with Gcc9 in Debug mode):

          5 - grisu-test (Child aborted)
          7 - format-test (Child aborted)
          8 - format-impl-test (Failed)
         12 - printf-test (Child aborted)

It trips this:

    fmt/include/fmt/core.h:266: assertion failed: negative value

Reply:
This suggests that the cast is misplaced.

Reply:
Yes, in fact it is not needed at all.

Reply:
Removing the cast also silences the warning, but still trips the assertion.

Reply:
Actually the static_cast<int64_t> is not needed.